### ISSUE TITLE
Using 2 or more images for training 

### ISSUE THREAD

**Issue by 17sarf** [pytorch/vision#1245]
I am working on a multispectral problem where i want to use the Faster R-CNN for pedestrian detection. I have been working with deep learning models now for a few months in PyTorch so I have a basic understanding of the architecture of deep learning models. What I would like to do is feed 2 images (e.g. RGB and thermal) with their annotations. So the output of my dataset class is visual_image, thermal_image, target, where target has the label and bounding box information. However, I am struggling to understand how to adjust the Faster R-CNN model so that I can feed in the 2 images and annotations.  This is is how I would do it for a basic model: [https://discuss.pytorch.org/t/halfway-fusion-with-two-datasets/42181 ](url).

 Any advice would be greatly appreciated and thank you in advance.

**Comment by fmassa** [comment]
One basic approach would be to concatenate the thermal image and the RGB image together, yielding an image with 4 channels, and change the first layer of resnet to take 4 channels instead of 3.
Other solutions are possible, but would require more changes.

Given that this is a modeling problem, it might be better suited for the discussion forums at http://discuss.pytorch.org

**Comment by 17sarf** [comment]
Thank you very much for your advice. I was actually looking into using halfway fusion, but I believe I may have come up with a solution.

**Comment by 17sarf** [comment]
I have managed to get it working using AlexNet. I have added a model (nn.Module) to concatenate the features from particular convoultional layers of AlexNet. However, I am having issues doing something similar for BackboneWithFPN for fasterrcnn_resnet50_fpn. This is what I did for AlexNet:

`class FusionModel(nn.Module):
    def __init__(self, net):
        super(FusionModel, self).__init__()
        self.net = net
        self.features1 = nn.Sequential(*list(self.net.children())[:4])
        self.features2 = nn.Sequential(*list(self.net.children())[4:])
        # self.conv = nn.Conv2d(1024, 512, kernel_size=1)
        self.conv = nn.Conv2d(384, 192, kernel_size=1)

    def forward(self, x, y):
        x = self.features1(x)
        y = self.features1(y)
        concat_features = torch.cat((x, y), dim=1)
        concat_features = self.conv(concat_features)
        output = self.features2(concat_features)
        return output

Although this could be applied to any model really. However, I would like to implement the FeaturePyramidNetwork as it seems to get me better results. Again, any advice would be greatly appreciated. Thank you in advance.

**Comment by fmassa** [comment]
@17sarf if you want to implement FPN, I'd recommend you modify the forward of your model to return an `OrderedDict` with the different activations for your model, and then use the `FeaturePyramidNetwork` module on top of that.

**Comment by 17sarf** [comment]
Thank you so much for your reply. I have attempted to adjust the forward function of ResNet in torchvision.models.resnet file. But it doesn't seem to be using that forward function. I'll continue to have a play around with it, but if you have any other suggestions, then that would be great.  I am still relatively new to implementing deep learning on pytorch, so perhaps I may be overlooking something simple.

**Comment by 17sarf** [comment]
@fmassa Thank you for advice. I now have the model working.
